### ISSUE TITLE
imshow() does not interpret aspect/extent when interpolation='none' in svg output

### ISSUE THREAD

**Issue by golobor** [matplotlib/matplotlib#7078]
imshow() ignores or misinterprets the `aspect` kwarg when `interpolation='none'`. The bug appears only in svg output, does not seem to be a problem in pdf or png output. The minimal code to reproduce the bug is below:

```python
import numpy as np
import matplotlib
matplotlib.use('SVG')
import matplotlib.pyplot as plt

mat = np.ones((40,20))

plt.figure()
plt.subplot(121)
plt.imshow(
    mat,
    aspect=0.5,
    interpolation='nearest')
plt.subplot(122)
plt.imshow(
    mat,
    aspect=0.5,
    interpolation='none')
plt.savefig('/tmp/test.svg')
```

An image of the output is attached below. The issue also appears when I try to use `extent` kwarg instead of `aspect`, i.e. see below (if needed, I can attach an image produced by the second example):

```python
plt.figure()
plt.subplot(121)
plt.imshow(
    mat,
    interpolation='nearest',
    extent=[0,20,20,0]
    
)
plt.subplot(122)
plt.imshow(
    mat,
    interpolation='none',
    extent=[0,20,20,0]
)
plt.savefig('/tmp/test2.svg')
```

![svg_aspect_interpolation](https://cloud.githubusercontent.com/assets/912906/18398220/c15c4496-7699-11e6-8d8b-612c8a3801be.png)

My python environment is:
Ubuntu 16.04, anaconda python distribution, python 3.5.2,  Matplotlib 1.5.1 (from conda)

**Comment by tacaswell** [comment]
Can you try with the 2.0.0b4 (`conda install -c conda-forge/label/rc -c conda-forge matplotlib` ?)

**Comment by golobor** [comment]
indeed, the problem doesn't occur in Matplotlib 2.0.0b4. Thank you!

**Comment by tacaswell** [comment]
Closing as fixed (and there should not be a 1.5.4).

**Comment by tacaswell** [comment]
@golobor  Thanks for taking the time to report this bug!

**Comment by Jeitan** [comment]
I know this is a year later, but can someone elucidate what went into the fixing of this bug?  I don't know why, but on a particular machine (which unfortunately does not have internet access) I am getting bad behavior just like this with matplotlib 2.0.0.  Which ... this bug fix should have gone into 2.0.0, yes?  But imshow is refusing to listen to 'aspect' when I set 'extent' (and interpolation is nearest, although I have not checked what changing that does).  What I have on the target machine is a vanilla install of Anaconda 4.3.0.  Unfortunately, I can't reproduce the behavior on *this* machine even by switching back to matplotlib 2.0.0, so I was hoping to find out where in the code I could look to see what's going on.

**Comment by tacaswell** [comment]
@Jeitan I did a quick search and did not find any obvious issue or PR that this is a duplicate of.   If you have a reproducible example you can try bisecting to find where it breaks or open a new issue.
